coral-hive/src/main/java/com/linkedin/coral/transformers/CoralRelToSqlNodeConverter.java

```diff
@@ -0,0 +1,357 @@
+
+    // projectResult's SqlNode representation: SELECT `complex`.`c` AS `col` FROM (VALUES  (0)) AS `t` (`ZERO`)
+    final Result projectResult = visitChild(0, e.getInput());
+
+    // Extract column(s) to unnest from projectResult
+    // to generate simpler operand for UNNEST operator
+    final List<SqlNode> unnestOperands = new ArrayList<>();
+
+    RelDataType recordType = null;
+    boolean withOrdinality = e.withOrdinality;
+
+    for (RexNode unnestCol : ((Project) e.getInput()).getChildExps()) {
+      unnestOperands.add(projectResult.qualifiedContext().toSql(null, unnestCol));
+      if (unnestCol.getType().getSqlTypeName().equals(SqlTypeName.ARRAY)
+          && unnestCol.getType().getComponentType().getSqlTypeName().equals(SqlTypeName.ROW)) {
+        recordType = unnestCol.getType().getComponentType();
+      }
+    }
+
+    // Generate SqlCall with Coral's UNNEST Operator and the unnestOperands. Also, persist ordinality and operand's datatype
+    final SqlNode unnestCall =
+        new CoralSqlUnnestOperator(withOrdinality, recordType).createCall(POS, unnestOperands.toArray(new SqlNode[0]));
+
+    // Append the alias to unnestCall by generating SqlCall with AS operator
+    List<SqlNode> asOperands = createAsFullOperands(e.getRowType(), unnestCall, projectResult.neededAlias);
+    final SqlNode aliasCall = SqlStdOperatorTable.AS.createCall(POS, asOperands);
+
+    // Append the LATERAL operator
+    final SqlNode lateralCall = SqlStdOperatorTable.LATERAL.createCall(POS, aliasCall);
+
+    // Reuse the same projectResult.neededAlias since that's already unique by directly calling "new Result(...)"
+    // instead of calling super.result(...), which will generate a new table alias and cause an extra
+    // "AS" to be added to the generated SQL statement and make it invalid.
+    return new Result(lateralCall, ImmutableList.of(Clause.FROM), null, e.getRowType(),
+        ImmutableMap.of(projectResult.neededAlias, e.getRowType()));
+  }
+
+  private void collectAliases(ImmutableMap.Builder<String, RelDataType> builder, SqlNode node,
+      Iterator<RelDataType> aliases) {
+    if (node instanceof SqlJoin) {
+      SqlJoin join = (SqlJoin) node;
+      collectAliases(builder, join.getLeft(), aliases);
+      collectAliases(builder, join.getRight(), aliases);
+    } else {
+      String alias = getAlias(node, -1);
+      assert alias != null;
+      builder.put(alias, aliases.next());
+    }
+  }
+
+  private String getAlias(SqlNode node, int ordinal) {
+    switch (node.getKind()) {
+      case AS:
+        return ((SqlCall) node).operand(1).toString();
+      case OVER:
+        return getAlias(((SqlCall) node).operand(0), ordinal);
+      case IDENTIFIER:
+        return Util.last(((SqlIdentifier) node).names);
+      case LATERAL:
+        SqlCall asNode = ((SqlCall) node).operand(0);
+        return getAlias(asNode, ordinal);
+      default:
+        return ordinal < 0 ? null : SqlUtil.deriveAliasFromOrdinal(ordinal);
+    }
```

Comment:
Could you please reference the Calcite methods those are copied from, and copy the comments too since they are helpful?

Reply:
resolved the issue by offline discussions. 
These overrides have been removed.